jms/src/main/scala/akka/stream/alpakka/jms/impl/JmsConnector.scala

```diff
@@ -232,6 +247,20 @@ private[jms] trait JmsConnector[S <: JmsSession] {
     allSessions.foreach(_.foreach(onSession.invoke))
   }
 
+  protected def closeConnection(connection: jms.Connection): Unit = {
+    // deregister exception listener to clear reference from JMS client to the Akka stage
+    Try(connection.setExceptionListener(null))
+    try {
+      connection.close()
+      log.info("JMS connection {} closed", connection)
+    } catch {
+      case NonFatal(e) => log.error(e, "Error closing JMS connection {}", connection)
+    }
+  }
+
+  protected def closeConnectionAsync(eventualConnection: Future[jms.Connection]): Future[Done] =
```

Comment:
I wonder what the consequences of closing connections asynchronously without blocking could be.  I realize that that was what was done before, but ideally, we would chain any subsequent operations after asynchronously closing the connection object with `AsyncCallbacks`.  Maybe it's not very important.